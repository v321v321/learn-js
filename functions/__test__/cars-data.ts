export type CarFuelType = 'Gasoline' | 'Diesel' | 'Hybrid' | 'Electric';

export interface Car {
  model: string;
  color: string;
  fuel: CarFuelType;
}

export const carsData: Car[] = [
  {
    model: 'Ferrari Cruze',
    color: 'orchid',
    fuel: 'Gasoline',
  },

  {
    model: 'Kia Challenger',
    color: 'fuchsia',
    fuel: 'Gasoline',
  },

  {
    model: 'Nissan Model X',
    color: 'gold',
    fuel: 'Diesel',
  },

  {
    model: 'Bentley Model S',
    color: 'purple',
    fuel: 'Hybrid',
  },

  {
    model: 'Land Rover CTS',
    color: 'ivory',
    fuel: 'Hybrid',
  },

  {
    model: 'Hyundai Civic',
    color: 'white',
    fuel: 'Hybrid',
  },

  {
    model: 'Honda Fortwo',
    color: 'mint green',
    fuel: 'Gasoline',
  },

  {
    model: 'Mazda Escalade',
    color: 'green',
    fuel: 'Gasoline',
  },

  {
    model: 'Lamborghini XC90',
    color: 'teal',
    fuel: 'Hybrid',
  },

  {
    model: 'Honda V90',
    color: 'lime',
    fuel: 'Hybrid',
  },

  {
    model: 'Volvo 2',
    color: 'olive',
    fuel: 'Gasoline',
  },

  {
    model: 'Jeep Beetle',
    color: 'lavender',
    fuel: 'Hybrid',
  },

  {
    model: 'Aston Martin Countach',
    color: 'tan',
    fuel: 'Gasoline',
  },

  {
    model: 'Mercedes Benz Golf',
    color: 'silver',
    fuel: 'Hybrid',
  },

  {
    model: 'Chevrolet Countach',
    color: 'teal',
    fuel: 'Diesel',
  },

  {
    model: 'Audi PT Cruiser',
    color: 'cyan',
    fuel: 'Diesel',
  },

  {
    model: 'Volkswagen PT Cruiser',
    color: 'ivory',
    fuel: 'Diesel',
  },

  {
    model: 'Ford Beetle',
    color: 'violet',
    fuel: 'Diesel',
  },

  {
    model: 'Nissan Wrangler',
    color: 'lime',
    fuel: 'Electric',
  },

  {
    model: 'Smart 2',
    color: 'sky blue',
    fuel: 'Hybrid',
  },

  {
    model: 'Maserati XTS',
    color: 'pink',
    fuel: 'Gasoline',
  },

  {
    model: 'Hyundai Impala',
    color: 'fuchsia',
    fuel: 'Diesel',
  },

  {
    model: 'Hyundai Prius',
    color: 'salmon',
    fuel: 'Hybrid',
  },

  {
    model: 'Kia XC90',
    color: 'indigo',
    fuel: 'Hybrid',
  },

  {
    model: 'Rolls Royce Civic',
    color: 'white',
    fuel: 'Gasoline',
  },

  {
    model: 'Mazda Focus',
    color: 'violet',
    fuel: 'Gasoline',
  },

  {
    model: 'Maserati Malibu',
    color: 'lavender',
    fuel: 'Hybrid',
  },

  {
    model: 'Aston Martin Model 3',
    color: 'black',
    fuel: 'Hybrid',
  },

  {
    model: 'Chrysler Civic',
    color: 'yellow',
    fuel: 'Diesel',
  },

  {
    model: 'Maserati Durango',
    color: 'orchid',
    fuel: 'Hybrid',
  },
];
