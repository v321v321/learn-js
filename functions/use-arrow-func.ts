// Написать функцию map, которая принимает на вход массив
// и стрелочную функцию, которая обрабатывает элементы массива

export const customMap = (arr, func) => {

  const arr2 = []

  arr.forEach(item => {

    arr2.push(func(item))

  })

  return arr2
}

// Написать функцию filter, которая принимает на вход массив
// и стрелочную функцию для фильтрации массива

export const customFilter = (arr, func) => {

  const arr2 = []

  arr.forEach(item => {

    if (func(item)) {
      arr2.push(item)
    }
  })

  return arr2
}

