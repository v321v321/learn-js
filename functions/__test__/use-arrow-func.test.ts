import { expect, test } from 'vitest';
import { customMap, customFilter } from '../examples/use-arrow-func';
import { Car, carsData } from './cars-data';

// Написать функцию map, которая принимает на вход массив
// и стрелочную функцию, которая обрабатывает элементы массива
test('Test custom map function', () => {
  expect(
    customMap(carsData, (car: Car) => ({
      ...car,
      model: car.model.toLocaleLowerCase(),
    }))
  ).toMatchSnapshot();
});

// Написать функцию filter, которая принимает на вход массив
// и стрелочную функцию для фильтрации массива
test('Test custom filter function', () => {
  expect(
    customFilter(carsData, (car: Car) => car.fuel === 'Diesel')
  ).toMatchSnapshot();
});
