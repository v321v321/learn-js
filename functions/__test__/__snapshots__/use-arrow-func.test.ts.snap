// Vitest Snapshot v1

exports[`Test custom filter function 1`] = `
[
  {
    "color": "gold",
    "fuel": "Diesel",
    "model": "Nissan Model X",
  },
  {
    "color": "teal",
    "fuel": "Diesel",
    "model": "Chevrolet Countach",
  },
  {
    "color": "cyan",
    "fuel": "Diesel",
    "model": "Audi PT Cruiser",
  },
  {
    "color": "ivory",
    "fuel": "Diesel",
    "model": "Volkswagen PT Cruiser",
  },
  {
    "color": "violet",
    "fuel": "Diesel",
    "model": "Ford Beetle",
  },
  {
    "color": "fuchsia",
    "fuel": "Diesel",
    "model": "Hyundai Impala",
  },
  {
    "color": "yellow",
    "fuel": "Diesel",
    "model": "Chrysler Civic",
  },
]
`;

exports[`Test custom map function 1`] = `
[
  {
    "color": "orchid",
    "fuel": "Gasoline",
    "model": "ferrari cruze",
  },
  {
    "color": "fuchsia",
    "fuel": "Gasoline",
    "model": "kia challenger",
  },
  {
    "color": "gold",
    "fuel": "Diesel",
    "model": "nissan model x",
  },
  {
    "color": "purple",
    "fuel": "Hybrid",
    "model": "bentley model s",
  },
  {
    "color": "ivory",
    "fuel": "Hybrid",
    "model": "land rover cts",
  },
  {
    "color": "white",
    "fuel": "Hybrid",
    "model": "hyundai civic",
  },
  {
    "color": "mint green",
    "fuel": "Gasoline",
    "model": "honda fortwo",
  },
  {
    "color": "green",
    "fuel": "Gasoline",
    "model": "mazda escalade",
  },
  {
    "color": "teal",
    "fuel": "Hybrid",
    "model": "lamborghini xc90",
  },
  {
    "color": "lime",
    "fuel": "Hybrid",
    "model": "honda v90",
  },
  {
    "color": "olive",
    "fuel": "Gasoline",
    "model": "volvo 2",
  },
  {
    "color": "lavender",
    "fuel": "Hybrid",
    "model": "jeep beetle",
  },
  {
    "color": "tan",
    "fuel": "Gasoline",
    "model": "aston martin countach",
  },
  {
    "color": "silver",
    "fuel": "Hybrid",
    "model": "mercedes benz golf",
  },
  {
    "color": "teal",
    "fuel": "Diesel",
    "model": "chevrolet countach",
  },
  {
    "color": "cyan",
    "fuel": "Diesel",
    "model": "audi pt cruiser",
  },
  {
    "color": "ivory",
    "fuel": "Diesel",
    "model": "volkswagen pt cruiser",
  },
  {
    "color": "violet",
    "fuel": "Diesel",
    "model": "ford beetle",
  },
  {
    "color": "lime",
    "fuel": "Electric",
    "model": "nissan wrangler",
  },
  {
    "color": "sky blue",
    "fuel": "Hybrid",
    "model": "smart 2",
  },
  {
    "color": "pink",
    "fuel": "Gasoline",
    "model": "maserati xts",
  },
  {
    "color": "fuchsia",
    "fuel": "Diesel",
    "model": "hyundai impala",
  },
  {
    "color": "salmon",
    "fuel": "Hybrid",
    "model": "hyundai prius",
  },
  {
    "color": "indigo",
    "fuel": "Hybrid",
    "model": "kia xc90",
  },
  {
    "color": "white",
    "fuel": "Gasoline",
    "model": "rolls royce civic",
  },
  {
    "color": "violet",
    "fuel": "Gasoline",
    "model": "mazda focus",
  },
  {
    "color": "lavender",
    "fuel": "Hybrid",
    "model": "maserati malibu",
  },
  {
    "color": "black",
    "fuel": "Hybrid",
    "model": "aston martin model 3",
  },
  {
    "color": "yellow",
    "fuel": "Diesel",
    "model": "chrysler civic",
  },
  {
    "color": "orchid",
    "fuel": "Hybrid",
    "model": "maserati durango",
  },
]
`;
